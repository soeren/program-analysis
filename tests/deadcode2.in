// Check for dead code elimination
// - Check intra-procedural dead code elimination
// - Check that unused function are removed

program

int x;
int y;

int never(int a) { // dead function
    int r;
    r := a*2;
    return r;
}

int stay(int b){
    int e;
    e := b * 12345; // dead statement
    e := b - 1;
    return e;
}

x := 11;
y := stay(x);

write(y);

end
