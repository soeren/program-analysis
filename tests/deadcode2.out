program
int x;
int y;
int stay(int b) // 4
{
     int e;
     e := b - 1; // 5
     return e; // 6
 }
x := 11; // 1
y := stay(x); // 2
write(y); // 3
end
